File: grokipedia_api/exceptions.py
"""Custom exceptions for Grokipedia API."""


class GrokipediaError(Exception):
    """Base exception for Grokipedia API errors."""
    pass


class GrokipediaNotFoundError(GrokipediaError):
    """Exception raised when a requested resource is not found."""
    pass


class GrokipediaAPIError(GrokipediaError):
    """Exception raised when there's an API-related error."""
    pass


class GrokipediaRateLimitError(GrokipediaError):
    """Exception raised when rate limit is exceeded."""
    pass

File: src/exceptions.ts
/**
 * Custom exceptions for Grokipedia API
 */

export class GrokipediaError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "GrokipediaError";
    Object.setPrototypeOf(this, GrokipediaError.prototype);
  }
}

export class GrokipediaNotFoundError extends GrokipediaError {
  constructor(message: string) {
    super(message);
    this.name = "GrokipediaNotFoundError";
    Object.setPrototypeOf(this, GrokipediaNotFoundError.prototype);
  }
}

export class GrokipediaAPIError extends GrokipediaError {
  constructor(message: string) {
    super(message);
    this.name = "GrokipediaAPIError";
    Object.setPrototypeOf(this, GrokipediaAPIError.prototype);
  }
}

export class GrokipediaRateLimitError extends GrokipediaError {
  constructor(message: string = "Rate limit exceeded") {
    super(message);
    this.name = "GrokipediaRateLimitError";
    Object.setPrototypeOf(this, GrokipediaRateLimitError.prototype);
  }
}
